src/views/collection-view/index.tsx

```diff
@@ -1,29 +1,108 @@
 import Link from 'next/link'
+import { TutorialLite as ClientTutorialLite } from 'lib/learn-client/types'
+import useCurrentPath from 'hooks/use-current-path'
+import { useCurrentProduct } from 'contexts'
 import SidebarSidecarLayout from 'layouts/sidebar-sidecar'
 import CoreDevDotLayout from 'layouts/core-dev-dot-layout'
-import { TutorialLite as ClientTutorialLite } from 'lib/learn-client/types'
-import ProductCollectionsSidebar from 'components/tutorials-sidebar/compositions/product-collections-sidebar'
-import { getTutorialSlug } from './helpers'
+import {
+  generateProductLandingSidebarNavData,
+  generateTopLevelSidebarNavData,
+} from 'components/sidebar/helpers'
+import TutorialsSidebar, {
+  HorizontalRule,
+  SectionList,
+  SectionTitle,
+} from 'components/tutorials-sidebar'
+import { CollectionCategorySidebarSection, getTutorialSlug } from './helpers'
 import { CollectionPageProps } from './server'
 import CollectionMeta from './components/collection-meta'
 
 function CollectionView({
   collection,
-  product,
   layoutProps,
 }: CollectionPageProps): React.ReactElement {
-  const { name, slug, description, shortName, tutorials } = collection
+  const currentProduct = useCurrentProduct()
+  const currentPath = useCurrentPath({ excludeHash: true, excludeSearch: true })
+  const { name, slug, description, tutorials } = collection
+
+  // TODO: refactor, very long
+  const sidebarNavDataLevels = [
+    {
+      levelButtonProps: {
+        levelDownButtonText: `${currentProduct.name} Home`,
+      },
+      menuItems: generateTopLevelSidebarNavData(),
+      showFilterInput: false,
+      title: 'Main Menu',
+    },
+    {
+      levelButtonProps: {
+        levelUpButtonText: 'Main Menu',
+        levelDownButtonText: 'Previous',
+      },
+      menuItems: generateProductLandingSidebarNavData(currentProduct),
+      showFilterInput: false,
+      title: currentProduct.name,
+    },
+    {
+      levelButtonProps: {
+        levelUpButtonText: `${currentProduct.name} Home`,
+        levelDownButtonText: 'Previous',
+      },
+      backToLinkProps: {
+        text: `${currentProduct.name} Home`,
+        href: `/${currentProduct.slug}`,
+      },
+      title: 'Tutorials',
+      children: [
+        <>
+          <SectionList
+            items={[
+              {
+                text: 'Overview',
+                href: `/${currentProduct.slug}/tutorials`,
+                isActive: currentPath === `/${currentProduct.slug}/tutorials`,
+              },
+            ]}
+          />
+          {layoutProps.collectionViewSidebarSections.map(
+            (section: CollectionCategorySidebarSection) => {
+              return (
+                <>
+                  <HorizontalRule />
+                  <SectionTitle text={section.title} />
+                  <SectionList items={section.items} />
+                </>
+              )
+            }
+          )}
+          <HorizontalRule />
```

Comment:
I liked your idea of separating this resources section into its own component. And ya perhaps we could separate this whole children piece into a component?

Reply:
Yeah... 🤔 I think the top two levels are the same for all of these views, so that could probably be shared between them with a helper function. And then each view could handle appending whatever extra levels it has. Thoughts on that?

Second thought would be to add something like a `TutorialsSidebarChildren` React component that lives in `TutorialsSidebar`'s file, and then replacing this React code under `children` with `<TutorialsSidebarChildren />` (and whatever props it needs). I think that would make it reusable in multiple views!

Reply:
I think that sounds great! Condensing these various children iterations into their own composed components, and having those live within the `TutorialSidebar` dir.